backend/tests/test_endpoints_moodle.py

```diff
@@ -0,0 +1,341 @@
+from fastapi.testclient import TestClient
+
+import pytest
+
+from rope.api.main import app
+from rope.api.database import SessionLocal
+from rope.api.sessions import get_request_session
+from rope.db.schema import CourseBuild, MoodleSetting, SchoolDistrict, UserAccount
+
+
+@pytest.fixture
+def test_client():
+    return TestClient(app)
+
+
+@pytest.fixture
+def db():
+    db = SessionLocal()
+    try:
+        yield db
+    finally:
+        db.close()
+
+
+@pytest.fixture(autouse=True)
+def clear_database_table(db):
+    db.query(CourseBuild).delete()
+    db.query(UserAccount).delete()
+    db.query(SchoolDistrict).delete()
+    db.query(MoodleSetting).delete()
+    db.commit()
+
+
+@pytest.fixture
+def setup_admin_session(mocker):
+    app.dependency_overrides[get_request_session] = override_admin_get_request_session
+    admin = {
+        "A1B2C3": {
+            "email": "admin@rice.edu",
+            "is_manager": False,
+            "is_admin": True,
+        }
+    }
+    mocker.patch(
+        "rope.api.sessions.session_store",
+        admin,
+    )
+
+
+@pytest.fixture
+def setup_manager_session(mocker):
+    app.dependency_overrides[get_request_session] = override_manager_get_request_session
+    manager = {
+        "Z9C8V7": {
+            "email": "manager@rice.edu",
+            "is_manager": True,
+            "is_admin": False,
+        }
+    }
+    mocker.patch(
+        "rope.api.sessions.session_store",
+        manager,
+    )
+
+
+@pytest.fixture
+def setup_nonadmin_authenticated_user_session(mocker):
+    app.dependency_overrides[get_request_session] = override_get_request_session
+    user = {
+        "12345": {
+            "email": "test@rice.edu",
+            "is_manager": False,
+            "is_admin": False,
+        }
+    }
+    mocker.patch(
+        "rope.api.sessions.session_store",
+        user,
+    )
+
+
+@pytest.fixture
+def create_course_build_setup_district(test_client, setup_admin_session):
+    new_school_district_data = {
+        "name": "snowfall_isd",
+        "active": True,
+    }
+    response = test_client.post(
+        "/admin/settings/district", json=new_school_district_data
+    )
```

Comment:
I happened to notice this and it may be a general pattern to address in a separate, dedicated issue: It's not ideal that we have fixtures that kind of rely on unrelated application behavior. For example, in this case the `/moodle/`tests fail if `/admin` endpoints aren't working. This could be avoided if the fixture injects into the DB like we do in (I think) most other cases.

Reply:
You are correct - I think it would be worth injecting into the DB as we do in other test cases.